File: src/main/java/com/restaurant/entity/Zone.java
package com.restaurant.entity;

import java.util.List;

public class Zone extends BaseEntity {
    private String nom;
    private double prixLivraison;
    private List<String> quartiers;
    
    public Zone() {}
    
    public Zone(String nom, double prixLivraison, List<String> quartiers) {
        this.nom = nom;
        this.prixLivraison = prixLivraison;
        this.quartiers = quartiers;
    }
    
    public String getNom() {
        return nom;
    }
    
    public void setNom(String nom) {
        this.nom = nom;
    }
    
    public double getPrixLivraison() {
        return prixLivraison;
    }
    
    public void setPrixLivraison(double prixLivraison) {
        this.prixLivraison = prixLivraison;
    }
    
    public List<String> getQuartiers() {
        return quartiers;
    }
    
    public void setQuartiers(List<String> quartiers) {
        this.quartiers = quartiers;
    }
    
    public boolean contientQuartier(String quartier) {
        return quartiers.contains(quartier);
    }
    
    @Override
    public String toString() {
        return "Zone{" +
               "id=" + getId() +
               ", nom='" + nom + '\'' +
               ", prixLivraison=" + prixLivraison +
               ", quartiers=" + quartiers +
               ", archived=" + isArchived() +
               '}';
    }
}

File: src/Entity/Zone.php
<?php

namespace App\Entity;

use App\Repository\ZoneRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ZoneRepository::class)]
#[ORM\Table(name: 'zones')]
class Zone
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 100, unique: true)]
    private ?string $nom = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?string $prixLivraison = null;

    #[ORM\Column(type: 'json')]
    private array $quartiers = [];

    #[ORM\Column(type: 'boolean')]
    private bool $archived = false;

    #[ORM\Column(type: 'datetime_immutable')]
    private \DateTimeImmutable $createdAt;

    #[ORM\Column(type: 'datetime_immutable')]
    private \DateTimeImmutable $updatedAt;

    #[ORM\OneToMany(mappedBy: 'zone', targetEntity: Commande::class)]
    private Collection $commandes;

    #[ORM\OneToMany(mappedBy: 'zone', targetEntity: Livreur::class)]
    private Collection $livreurs;

    public function __construct()
    {
        $this->createdAt = new \DateTimeImmutable();
        $this->updatedAt = new \DateTimeImmutable();
        $this->commandes = new ArrayCollection();
        $this->livreurs = new ArrayCollection();
    }

    // Getters and setters...
    public function getId(): ?int { return $this->id; }
    public function getNom(): ?string { return $this->nom; }
    public function setNom(string $nom): self { $this->nom = $nom; return $this; }
    public function getPrixLivraison(): ?string { return $this->prixLivraison; }
    public function setPrixLivraison(string $prixLivraison): self { $this->prixLivraison = $prixLivraison; return $this; }
    public function getQuartiers(): array { return $this->quartiers; }
    public function setQuartiers(array $quartiers): self { $this->quartiers = $quartiers; return $this; }
    public function isArchived(): bool { return $this->archived; }
    public function setArchived(bool $archived): self { $this->archived = $archived; return $this; }
    public function getCreatedAt(): \DateTimeImmutable { return $this->createdAt; }
    public function setCreatedAt(\DateTimeImmutable $createdAt): self { $this->createdAt = $createdAt; return $this; }
    public function getUpdatedAt(): \DateTimeImmutable { return $this->updatedAt; }
    public function setUpdatedAt(\DateTimeImmutable $updatedAt): self { $this->updatedAt = $updatedAt; return $this; }
}